server/app/controllers/announcements-controller.ts

```diff
@@ -0,0 +1,72 @@
+import { RequestHandler } from "express";
+import { Announcement } from "../../types/announcements-types"; 
+import announcementsService from "../services/announcements-service";
+
+const getAll: RequestHandler<
+  never,
+  Announcement[] | { error: string },
+  never
+> = async (req, res) => {
+  try {
+    const resp = await announcementsService.getAll();
+    res.status(200).json(resp);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+};
+
+const post: RequestHandler<
+  never,
+  { id: number } | { error: string },
+  Announcement
+> = async (req, res) => {
+  try {
+    const resp = await announcementsService.insert(req.body);
+    res.status(201).json(resp);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+};
+const remove: RequestHandler<
+  { id: string },
+  Response | { error: string } | { message: string },
+  never
+> = async (req, res) => {
+  try {
+    const rowCount = await announcementsService.remove(req.params.id);
+    if (!rowCount) {
+      return res.status(400).json({ error: "Record not found" });
+    }
+    res.sendStatus(204);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+};
+const update: RequestHandler<
+  { id: number },
+  { error: string } | { message: string } | Announcement,
+  Announcement
+> = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { title, description, is_enabled } = req.body; 
+    const resp = await announcementsService.update({id: Number(id),
```

Comment:
can we pass `id` as `string` here and convert it to a number in the service like how we're doing it [here](https://github.com/hackforla/food-oasis/blob/e75277a1cfb6f0bc1b6575b4370dff2b50eb8fe8/server/app/services/faq-service.ts#L73). Just want us to be more consistent 😅

Reply:
Thank you. Made it like faq-service. Hope that is ok.